Binding flow

diivm opened the pull request:
WIP

diivm committed:
[cmake_frontend.py] add get_library_targets; add some documentation

diivm committed:
[cmake_frontend.py] change parameter optionality; ret type of functions

- makes easier for the user, doesn't have to access dict values in the case of single use.

diivm committed:
[bind.py] bind.py

diivm committed:
[cmake_frontend.py] change parameter optionality; return type

of function `get_compilation_arguments`

kunaltyagi commented on clang_bind/cmake_frontend.py:
```diff
@@ -220,29 +223,35 @@ def _set_targets_from_codemodel(self):
-            for target in targets
-        }
+        library_targets_objs = filter(
+            lambda target: target.get_type() == "SHARED_LIBRARY", self.targets.values()
```
What about STATIC build?

diivm replied:
Didn't see it in `pcl`, so didn't get to test how it is.

kunaltyagi replied:
`PCL_SHARED_LIBS` can be used to choose between shared and static library for pcl

diivm replied:
Can you please provide a reference link or example?

kunaltyagi replied:
`cmake -DPCL_SHARED_LIBS:BOOL=FALSE` or similar

diivm committed:
[parse.py] add support for inclusion sources

- `is_cursor_in_file` -> `is_cursor_in_files`: support multiple files
- update `_is_valid_child` to also check for inclusion sources too

diivm committed:
fixup! [bind.py] bind.py

diivm committed:
fixup! [bind.py] bind.py

diivm committed:
fixup! [bind.py] bind.py

diivm committed:
[init_bindings.sh] add file
